merlin/study/status.py

```diff
@@ -780,45 +783,98 @@ def _remove_steps_without_statuses(self):
 
         self.requested_statuses = result
 
-    def apply_filters(self, filter_types: List[str], filters: List[str]):
+    def _search_for_filter(self, filter_to_apply: List[str], entry_to_search: Union[List[str], str]) -> bool:
         """
-        Given a list of filters, filter the dict of requested statuses by them.
+        Search an entry to see if our filter(s) apply to this entry. If they do, return True. Otherwise, False.
 
-        :param `filter_types`: A list of str denoting the types of filters we're applying
-        :param `filters`: A list of filters to apply to the dict of statuses we read in
+        :param filter_to_apply: A list of filters to search for
+        :param entry_to_search: A list or string of entries to search for our filters in
+        :returns: True if a filter was found in the entry. False otherwise.
         """
-        LOG.info(f"Filtering tasks using these filters: {filters}")
+        if not isinstance(entry_to_search, list):
+            entry_to_search = [entry_to_search]
 
-        # Create a deep copy of the dict so we can make changes to it while we iterate
-        result = deepcopy(self.requested_statuses)
+        filter_matches = []
+        apply_list_of_regex(filter_to_apply, entry_to_search, filter_matches, display_warning=False)
+        if len(filter_matches) != 0:
+            return True
+        return False
 
+    def apply_filters(self):
+        """
+        Apply any filters given by the --workers, --return-code, and/or --task-status arguments.
+        This function will also apply the --max-tasks limit if it was set by a user. We apply this
+        limit here so it can be done in-place; if we called apply_max_tasks_limit instead, this
+        would become a two-pass algorithm and can be really slow with lots of statuses.
+        """
+        if self.args.max_tasks is not None:
+            # Make sure the max_tasks variable is set to a reasonable number and store that value
+            if self.args.max_tasks > self.num_requested_statuses:
+                LOG.debug(
```

Comment:
I feel like this should maybe be a different log level. Warning, perhaps?

Reply:
yes you're right definitely a warning rather than debug